src/compiler/scala/tools/nsc/classpath/DirectoryClassPath.scala

```diff
@@ -119,6 +121,55 @@ trait JFileDirectoryLookup[FileEntryType <: ClassRepresentation] extends Directo
   def asClassPathStrings: Seq[String] = Seq(dir.getPath)
 }
 
+object JImageDirectoryLookup {
+  import java.nio.file._, java.net.URI, scala.collection.JavaConverters._
+  def apply(): List[ClassPath] = {
+    try {
+      val fs = FileSystems.getFileSystem(URI.create("jrt:/"))
+      val dir: Path = fs.getPath("/modules")
```

Comment:
btw, where's the spec for `jrt:/modules/...`? in http://openjdk.java.net/jeps/220 i could only find 

> A jrt URL is a hierarchical URI, per RFC 3986, with the syntax `jrt:/[$MODULE[/$PATH]]`

and

> `jrt:/` refers to the entire collection of class and resource files stored in the current run-time image.

example: `jrt:/java.base/java/lang/Class.class`

---

OK, Google. https://bugs.openjdk.java.net/browse/JDK-8066492